command/job_stop.go

```diff
@@ -138,29 +144,62 @@ func (c *JobStopCommand) Run(args []string) int {
 		return 1
 	}
 
-	// Confirm the stop if the job was a prefix match.
-	if jobID != *job.ID && !autoYes {
-		question := fmt.Sprintf("Are you sure you want to stop job %q? [y/N]", *job.ID)
+	getConfirmation := func(question string) (int, bool) {
 		answer, err := c.Ui.Ask(question)
 		if err != nil {
 			c.Ui.Error(fmt.Sprintf("Failed to parse answer: %v", err))
-			return 1
+			return 1, false
 		}
 
 		if answer == "" || strings.ToLower(answer)[0] == 'n' {
 			// No case
 			c.Ui.Output("Cancelling job stop")
-			return 0
+			return 0, false
 		} else if strings.ToLower(answer)[0] == 'y' && len(answer) > 1 {
 			// Non exact match yes
 			c.Ui.Output("For confirmation, an exact ‘y’ is required.")
-			return 0
+			return 0, false
 		} else if answer != "y" {
 			c.Ui.Output("No confirmation detected. For confirmation, an exact 'y' is required.")
-			return 1
+			return 1, false
+		}
+		return 0, true
+	}
+
+	// Confirm the stop if the job was a prefix match
+	if jobID != *job.ID && !autoYes {
+		question := fmt.Sprintf("Are you sure you want to stop job %q? [y/N]", *job.ID)
+		code, confirmed := getConfirmation(question)
+		if !confirmed {
+			return code
 		}
 	}
 
+	// Confirm we want to stop only a single region of a multiregion job
+	if job.IsMultiregion() && !global {
+		question := fmt.Sprintf(
+			"Are you sure you want to stop multi-region job %q in a single region? [y/N]", *job.ID)
+		code, confirmed := getConfirmation(question)
+		if !confirmed {
+			return code
+		}
+	}
+
+	// scatter-gather job stop for multi-region jobs
+	if global && job.IsMultiregion() {
+		for _, region := range job.Multiregion.Regions {
+			// Invoke the stop
+			wq := &api.WriteOptions{Namespace: jobs[0].JobSummary.Namespace, Region: region.Name}
+			evalID, _, err := client.Jobs().Deregister(*job.ID, purge, wq)
+			if err != nil {
+				c.Ui.Error(fmt.Sprintf("Error deregistering job in %q: %s", region.Name, err))
+				return 1
+			}
+			c.Ui.Output(evalID)
+		}
+		return 0
+	}
+
 	// Invoke the stop
 	wq := &api.WriteOptions{Namespace: jobs[0].JobSummary.Namespace}
 	evalID, _, err := client.Jobs().Deregister(*job.ID, purge, wq)
```

Comment:
is this still necessary?

Reply:
That's where single-region jobs get stopped.

Reply:
ah sorry, didn't see return 0 in the multiregion block :(